rewrite-java/src/main/java/org/openrewrite/java/ReplaceAnnotation.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class ReplaceAnnotation extends Recipe {
+    @Option(displayName = "Annotation pattern to replace",
+            description = "An annotation pattern, expressed as a method pattern to replace.",
```

Comment:
I was going to request a link to the corresponding documentation, but couldn't find it. @mike-solomon do you know if we've documented annotation matchers?

Reply:
I’m not familiar with the term annotation matcher - but maybe it’s similar to a method pattern? If not, I don’t think we have any docs on it. 

https://docs.openrewrite.org/reference/method-patterns

Reply:
Thanks for confirming. It is indeed very similar in concept. But rather than matching method calls (or declarations) its purpose is to match annotations.

Reply:
Yes it is similar and are both based on AspectJ Pointcut expressions. Would it be sufficient to create a documentation issue for AnnotationMatcher and mention this description?

Reply:
Yes, I believe we can just change the description now to mention that it is an annotation matcher and then when we get around to documenting that we can insert a link here.